src/Microsoft.AspNetCore.SignalR.Protocols.Json/Protocol/JsonHubProtocol.cs

```diff
@@ -197,13 +200,26 @@ private HubMessage ParseMessage(Utf8BufferTextReader textReader, IInvocationBind
 
                                         if (string.IsNullOrEmpty(invocationId))
                                         {
-                                            // If we don't have an invocation id then we need to store it as a JToken so we can parse it later
+                                            // If we don't have an invocation id we can't look up the type now
+                                            // need to store it as a JToken so we can parse it later
                                             itemToken = JToken.Load(reader);
+                                            break;
                                         }
-                                        else
+
+                                        var itemType = binder.GetStreamItemType(invocationId);
```

Comment:
We know it's a stream, because it has an `item` field and so is a stream item. If we added a field named `item` to a non-streaming message this would break though.